Gems/Terrain/Assets/Shaders/Terrain/TerrainPBR_ForwardPass.azsl

```diff
@@ -80,26 +80,38 @@ ForwardPassOutput TerrainPBR_MainPassPS(VSOutput IN)
     
     // ------- Macro Color / Normal -------
     float3 macroColor = TerrainMaterialSrg::m_baseColor.rgb;
-    [unroll] for (uint i = 0; i < 4 && (i < ObjectSrg::m_macroMaterialCount); ++i)
+
+    // There's a bug that shows up with an NVidia GTX 1660 Super card happening on driver versions as recent as 496.49 (10/26/21) in which
+    // the IN.m_uv values will intermittently "flicker" to 0.0 after entering and exiting game mode.  
+    // (See https://github.com/o3de/o3de/issues/5014)
+    // This bug has only shown up on PCs when using the DX12 RHI.  It doesn't show up with Vulkan or when capturing frames with PIX or
+    // RenderDoc.  Our best guess is that it is a driver bug.  The workaround is to use the IN.m_uv values in a calculation prior to the 
+    // point that we actually use them for macroUv below.  The "if(any(!isnan(IN.m_uv)))" seems to be sufficient for the workaround.  The
+    // if statement will always be true, but just the act of reading these values in the if statement makes the values stable.  Removing
+    // the if statement causes the flickering to occur using the steps documented in the bug.
+    if (any(!isnan(IN.m_uv)))
```

Comment:
Hmm. Not sure if there is a perf cost to it but to me using 'all' instead of 'any' reads a lot better. Basically currently if lets say IN.m_uv.x is not nan but IN.m_uv.y is nan it will go into the 'if' statement. Using 'all' just makes more logical sense even though this is just to fix a driver bug.

Reply:
Since we're not actually trying to guard against NaNs here, and there's really no way it could/should ever be a NaN, I was intentionally choosing the (theoretically) slightly faster "any".  This check really isn't intended to make the code any more robust, it's just meant to be the most minimal workaround possible.  I'd prefer to leave it as-is unless you really strongly object.

Reply:
Was not a strong objection :)